src/nupic/algorithms/TemporalMemory.cpp

```diff
@@ -899,12 +899,12 @@ void TemporalMemory::setPredictedSegmentDecrement(Permanence predictedSegmentDec
   predictedSegmentDecrement_ = predictedSegmentDecrement;
 }
 
-Permanence TemporalMemory::getMaxSegmentsPerCell() const
+UInt TemporalMemory::getMaxSegmentsPerCell() const
```

Comment:
I'd prefer if we use `UInt32` here and elsewhere to avoid differences due to the double precision option.